staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -140,6 +141,16 @@ func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
 		clientConfig.Timeout = timeout
 	}
 
+	if configClusterInfo.ConnectionTimeout != "" {
+		connectionTimeout, err := ParseTimeout(configClusterInfo.ConnectionTimeout)
+		if err != nil {
+			return nil, err
+		}
+		clientConfig.ConnectionTimeout = connectionTimeout
+	} else {
+		clientConfig.ConnectionTimeout = 7 * time.Second
```

Comment:
What would you suggest? While in my environment I am of course aware of this issue and happy to configure this explicitly, leaving users who are not aware of it open to this issue does not seem like a "sane default" given the severity of the problem.